test/dynamo/test_repros.py

```diff
@@ -2065,6 +2065,56 @@ def fn(x):
         self.assertEqual(cnt.frame_count, 1)
         self.assertEqual(cnt.op_count, 1)
 
+    def test_nested_while_loop_graph_break(self):
+        def inner_loop(x):
+            i = 3
+            while i > 0:
+                i -= 1
+                x += 1
+                torch._dynamo.graph_break()
+            return x
+
+        def inner(x):
+            inner_loop(x)
+            return torch.sin(x)
+
+        def fn(x):
+            i = 20
+            while i > 10:
+                x = inner(x)
+                i -= 1
+                torch._dynamo.graph_break()
+            return x
+
+        cnt = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(cnt)(fn)
+        x = torch.randn(4)
+        opt_fn(x)
+        self.assertEqual(cnt.frame_count, 1)
+        self.assertEqual(cnt.op_count, 1)
```

Comment:
This test is so useful and common we could have a helper for it

```python
def assert_no_graph_break(fn, inputs):
    cnt = torch._dynamo.testing.CompileCounter()
    opt_fn = torch._dynamo.optimize(cnt)(fn)
    opt_fn(inputs)
    self.assertEqual(cnt.frame_count, 1)
    self.assertEqual(cnt.op_count, 1)
```

Reply:
Agreed that would be a good refactor

Reply:
Agree this is nice thing to do. I (or anyone interested) could do it in a separate PR.